tests/test_optional_extras.py

```diff
@@ -0,0 +1,70 @@
+from app.models.goal import Goal
+from app.models.task import Task
+from datetime import datetime
+import pytest
+
+
+def test_get_tasks_for_specific_invalid_goal(client):
+    # Act
+    response = client.get("/goals/abc/tasks")
+    response_body = response.get_json()
+
+    # Assert
+    assert response.status_code == 400
+    assert response_body == {"details":f"Invalid {Goal.__name__} id"}
+
+    assert Goal.query.all() == []
+
+
+
+def test_get_tasks_for_specific_goal_no_goal(client):
+    # Act
+    response = client.get("/goals/1/tasks")
+    response_body = response.get_json()
+
+    # Assert
+    assert response.status_code == 404
+    assert response_body == {"details":f"{Goal.__name__} not found"}
+
+
+def test_create_task_with_date(client):
+    # Act
+    response = client.post("/tasks", json={
+        "title": "A Brand New Task",
+        "description": "Test Description",
+        "datetime": "07/12/99"
```

Comment:
Ah, your earlier code around the `"datetime"` key makes a bit more sense to me now, but I think just as an API design `"datetime"` as a key name could be better saying what the key _represents_ rather than what _type of data_ the key is. That is, something like `"completed_date"` would get more information about the purpose of this key while still making clear you only accept dates.